### DISCUSSION TITLE
How to get List of Interface object while interface having mutliple implementation.

### DISCUSSION THREAD

**Discussion by woodsongem** [AutoFixture/AutoFixture#1363] [General]
Hello All,

We have scenarios like below and I am trying to get an instance of List<A1> object. How to do it ?

interface A1 { }
class B1: A1{}
class C1: A1 {}
class D1: A1{}

**Comment by Kralizek** [comment]
What would you like to achieve? A list of A1 filled with a mix of instances of B1, C1, D1?

**Reply by woodsongem** [reply]
trying to create List<A1> mock objects which include A1=>B1, A1=>C1, A1=>D1  and passing class constructor.  inside class loop through list and calling method. have to mock some method in the interface.

**Reply by GeraldLx** [reply]
I think we need more input to help here.

The idea behind AutoFixture is to avoid repetitive code in tests and make tests more readable.

Your wish can be achieved as sample as:
var list = new List<A1> { fixture.Create<B1>(), fixture.Create<C1>(), fixture<D1>() }

If you want to write a test that shall work with A1 instances, regardless of implementation, you can of course write a specimenbuilder picking randomly among B1, C1 or D1.

Or perhaps you want to test all your A1 implementations, then it will be probably best to just get all types via definition or reflection. But this has nothing to do with AutoFixture.

Also your test and mocking the methods is more dependent on your implementation (are the methods virtual?) and your mocking framework, than AutoFixture.
